packages/preferences-api/src/browser/preference-proxy.ts

```diff
@@ -26,48 +26,59 @@ export interface PreferenceEventEmitter<T> {
     const toDispose = new DisposableCollection();
     const onPreferenceChangedEmitter = new Emitter<PreferenceChange>();
     toDispose.push(onPreferenceChangedEmitter);
-    toDispose.push(preferences.onPreferenceChanged(e => {
-        if (e.preferenceName in schema.properties) {
-            if (e.newValue) {
+    toDispose.push(preferences.onPreferenceChanged(change => {
+        if (change.preferenceName in schema.properties) {
+            /* Can't simply use change.newValue as this could be a boolean */
+            if (change.newValue !== undefined) {
                 // Fire the pref if it's valid according to the schema
                 if (validatePreference(schema, {
-                    [e.preferenceName]: e.newValue
+                    [change.preferenceName]: change.newValue
                 })) {
-                    onPreferenceChangedEmitter.fire(e);
+                    onPreferenceChangedEmitter.fire(change);
                 } else {
                     // Fire the default preference
                     onPreferenceChangedEmitter.fire({
-                        preferenceName: e.preferenceName,
-                        newValue: configuration[e.preferenceName]
+                        preferenceName: change.preferenceName,
+                        newValue: schema.properties[change.preferenceName].default || undefined
                     });
                 }
             } else {
-                // TODO If it's deleted, fire the default preference
-                onPreferenceChangedEmitter.fire(e);
+                /* Deleted preference, fire the default preference */
+                onPreferenceChangedEmitter.fire({
+                    preferenceName: change.preferenceName,
+                    newValue: schema.properties[change.preferenceName].default || undefined,
+                    oldValue: change.oldValue
+                });
             }
         }
     }));
+
+    /* Create a targer object with only the property names as properties, removing "type" and other stuff from the schema */
+    const targetConfig: Configuration = {};
+    Object.keys(schema.properties).forEach(prop => {
+        targetConfig[prop];
+    });
     // tslint:disable-next-line:no-any
-    return new Proxy(configuration as any, {
+    return new Proxy((targetConfig as any), {
         get: (_, p: string) => {
             if (p in schema.properties) {
-                if (p in configuration) {
-                    const preference = preferences.get(p, configuration[p]);
+                const preference = preferences.get(p, schema.properties[p].default);
+                if (preference) {
                     if (validatePreference(schema, {
                         [p]: preference
                     })) {
                         return preference;
                     } else {
-                        return configuration[p];
+                        return schema.properties[p].default || undefined;
```

Comment:
I wonder why `|| undefined`, I've seen it above as well, seems to be redundant as glance

Reply:
I think not all preferences in the schema have the "default" property, so that's why if it didn't exist returning undefined would make sense.

Reply:
if `default` is not present then the value of `default` is `undefined`, is not it?

Reply:
Yes you're right, I don't know why I was under the impression that this would throw an error, removed.